logstash-core/lib/logstash/patches/resolv.rb

```diff
@@ -0,0 +1,36 @@
+require "resolv"
+
+# ref: https://github.com/logstash-plugins/logstash-filter-dns/issues/40
+#
+# JRuby 9k versions prior to 9.1.16.0 have a bug which crashes IP address
+# resolution after 64k unique IP addresses resolutions.
+#
+# Note that the oldest JRuby version in LS 6 is 9.1.13.0 and
+# JRuby 1.7.25 and 1.7.27 (the 2 versions used across LS 5) are not affected by this bug.
+
+# make sure we abort if a known correct JRuby version is installed 
+# to avoid having an unnecessary legacy patch being applied in the future.
+raise("Unnecessary patch on resolv.rb for JRuby version 9.1.16+") if Gem::Version.new(JRUBY_VERSION) >= Gem::Version.new("9.1.16.0")
+
+# The code below is copied from JRuby 9.1.16.0 resolv.rb:
+# https://github.com/jruby/jruby/blob/9.1.16.0/lib/ruby/stdlib/resolv.rb#L775-L784
+
```

Comment:
Include copyright and license from https://github.com/jruby/jruby/blob/9.1.16.0/COPYING and explicitly indicate that we are redistribution under EPL:

~~~ ruby
# Copyright (c) 2007-2018 The JRuby project.
# Redistributed under the terms of the Eclipse Public License.
~~~

Reply:
sure. but this is an interesting case of including JRuby code for the purpose of patching JRuby itself using metaprogramming (not really for executing that code within LS) for which we already implicitly respect its copyright+license since we already include it as source code and the license terms have not changed between .13 and .16.